Alicerce/Sources/Network/Mappable.swift

```diff
@@ -0,0 +1,41 @@
+//
+//  Mappable.swift
+//  Alicerce
+//
+//  Created by Luís Afonso on 06/04/2017.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public enum MappableError: Error {
+
+    case custom(String)
+
+    /// Unwrapps the enum custom value
+    var description: String {
+        switch self {
+        case let .custom(description): return description
+        }
+    }
+}
+
+public protocol Mappable {
+
+    /// Converts the object from a `Any` (Dictionary) to Itself.
+    ///
+    /// - Parameter object: 
+    /// An object that information to fill the model. Usually a Dictionary.
+    /// - Returns: 
+    /// The object itself filled with values from the object
+    /// - Throws: 
+    /// A MappableError
+    static func model(from object: Any) throws -> Self
+
+    /// Converts Itself into a `AnyObject` (Dictionary) representation
+    /// You can simply return `NSNull()` if it doesn't make sense in your context to do that.
+    ///
+    /// - Returns: 
+    /// An `AnyObject` (like a `Dictionary` representation)
+    func json() -> AnyObject
```

Comment:
Thats why I was in two minds about it. Part of me thought it would go well as a computed property due to 0 arguments, however it does act like a function, which means maybe a property is not the way. I'm happy with keeping it as a function :smile: